gradle/publish.gradle

```diff
@@ -96,6 +98,43 @@ configure(javaMainProjects) { project ->
     }
 }
 
+configure(scalaProjects) { project ->
+    apply plugin: 'maven-publish'
+    apply plugin: 'signing'
+
+    task sourcesJar(type: Jar) {
+        from project.sourceSets.main.allScala
+        classifier = 'sources'
+    }
+
+    task scaladocJar(type: Jar) {
+        from scaladoc
+        classifier = 'scaladoc'
+    }
+
+    publishing {
+        publications {
+            mavenJava(MavenPublication) {
+                artifactId = project.archivesBaseName
+                from project.components.java
+                artifact sourcesJar
+                if (!scalaVersion.startsWith("2.13")) {
+                    artifact scaladocJar  // https://github.com/gradle/gradle/issues/9855
```

Comment:
Not really this just disables it for scala 2.13.

We still need different tasks for the docs and sources because one needs to hook into scaladoc and the other uses the allScala source not allJava.  The differences are very small though.

Reply:
Added: JAVA-3435